server/src/test/java/org/opensearch/action/admin/cluster/node/stats/NodeStatsTests.java

```diff
@@ -1065,4 +1083,257 @@ private static RemoteTranslogTransferTracker.Stats getRandomRemoteTranslogTransf
+        commonStatsFlags.set(CommonStatsFlags.Flag.Indexing, true);
+        commonStatsFlags.setAggregateNodeIndicesStatsResponsesOnLevel(true);
+
+        levelParams.forEach(levelParam -> {
+            ArrayList<String> level_arg = new ArrayList<>();
+            level_arg.add(levelParam.getRestName());
+
+            commonStatsFlags.setLevels(level_arg.toArray(new String[0]));
+
+            Index newIndex = new Index("index", "_na_");
+
+            MockNodeIndicesStats mockNodeIndicesStats = generateMockNodeIndicesStats(commonStats, newIndex, commonStatsFlags);
+
+            // To test out scenario when the incoming node stats response is from a node with an older ES Version.
+            try (BytesStreamOutput out = new BytesStreamOutput()) {
+                mockNodeIndicesStats.writeTo(out);
+                try (StreamInput in = out.bytes().streamInput()) {
+                    MockNodeIndicesStats newNodeIndicesStats = new MockNodeIndicesStats(in);
+                    switch (levelParam) {
+                        case NODE:
+                            assertNull(newNodeIndicesStats.getStatsByIndex());
+                            assertNull(newNodeIndicesStats.getStatsByShard());
+                            break;
+                        case INDICES:
+                            assertNull(newNodeIndicesStats.getStatsByShard());
+                            assertNotNull(newNodeIndicesStats.getStatsByIndex());
+                            break;
+                        case SHARDS:
+                            assertNull(newNodeIndicesStats.getStatsByIndex());
+                            assertNotNull(newNodeIndicesStats.getStatsByShard());
+                            break;
+                    }
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    public void testNodeIndicesStatsToXContent() {
+        long numDocs = randomLongBetween(0, 10000);
+        long numDeletedDocs = randomLongBetween(0, 100);
+        List<NodeIndicesStats.Fields> levelParams = new ArrayList<>();
+        levelParams.add(NodeIndicesStats.Fields.INDICES);
+        levelParams.add(NodeIndicesStats.Fields.SHARDS);
+        levelParams.add(NodeIndicesStats.Fields.NODE);
+        CommonStats commonStats = new CommonStats(CommonStatsFlags.NONE);
+
+        commonStats.docs = new DocsStats(numDocs, numDeletedDocs, 0);
+        commonStats.store = new StoreStats(100, 0L);
+        commonStats.indexing = new IndexingStats();
+
+        CommonStatsFlags commonStatsFlags = new CommonStatsFlags();
+        commonStatsFlags.clear();
+        commonStatsFlags.set(CommonStatsFlags.Flag.Docs, true);
+        commonStatsFlags.set(CommonStatsFlags.Flag.Store, true);
+        commonStatsFlags.set(CommonStatsFlags.Flag.Indexing, true);
+        commonStatsFlags.setAggregateNodeIndicesStatsResponsesOnLevel(true);
```

Comment:
Can we have a test which compares the toXContent of aggregate response with that of the non-aggregate one. Assert if the responses are identical.

Reply:
Added the test - `testNodeIndicesStatsWithAndWithoutAggregations` on next revision